### ISSUE TITLE
Conditionals for file creation

### ISSUE THREAD

**Issue by nrinaudo** [foundweekends/giter8#345]
The current version supports conditionals *inside* files, but not *about* files.
That is, I can include or exclude blocks of text depending on the truthiness of a value. I cannot, however, create files only if a certain value is truthy.

This would be hugely helpful in at least one case: the template I'm working on will support multiple deployment options (raw JAR, AWS, debian package...). Each method needs different files to be present in different paths. 

Currently, I can either include everything and ask developers to manually delete what's irrelevant, or include nothing and point them to a repo that contains the right files.

Ideally, they'd just answer a question (`use AWS? [YES/no]`) and everything would be set up.

**Comment by laughedelic** [comment]
> Currently, I can either include everything and ask developers to manually delete what's irrelevant, or include nothing and point them to a repo that contains the right files.

I think you have a third option: use [scaffolds](http://www.foundweekends.org/giter8/scaffolding.html) (sub-templates).

**Comment by nrinaudo** [comment]
Interesting. I think I completely misunderstood what scaffolds were, if you feel they apply to my use case. I'll look into it, thanks!

**Comment by laughedelic** [comment]
I think that the docs are confusing and I haven't used it myself yet. But if I understand it correctly, it's just sub-templates that live inside your project and you can use a special scaffold plugin to apply them.

**Comment by nrinaudo** [comment]
Right, but my understanding is that the scaffold plugin needs to live in the project generated by the initial g8 template. My ideal solution would be to generate a usable project, not to use a template to generate a template to generate a project.

But as I said, I might have misunderstood the point of scaffolds and will look into them.

**Comment by pwood** [comment]
I too would appreciate this feature, I've had a look at scaffold and including g8 content in the rendered template isn't ideal. As I would also then have to have a manual instruction step to remove the unneeded g8 content.

My use case is very similar to @nrinaudo, we want to be able to toggle large features on and off at initial creation.

**Comment by avboy72** [comment]
Add me to the list.

**Comment by notmu** [comment]
I have accomplished this in the following way if it helps @avboy72 @pwood @nrinaudo 
For context this is done in a Scala cross project. I'll throw the g8 up [here](https://github.com/notmu/cross.g8) when I get a chance.. ✌️ 
```bash
src/main/g8
├── $name__normalize$
│   ├── $if(jvm.truthy)$jvm$endif$
│   │   └── src
│   │       └── main
│   │           └── scala
│   │               └── $organization__packaged$
│   │                   └── $name;format="Camel"$.scala
│   ├── $if(native.truthy)$native$endif$
│   │   └── src
│   │       └── main
│   │           └── scala
│   │               └── $organization__packaged$
│   │                   └── $package$
│   │                       └── $name;format="Camel"$.scala
│   ├── $if(scalajs.truthy)$js$endif$
│   │   └── src
│   │       └── main
│   │           └── scala
│   │               └── $organization__packaged$
│   │                   └── $name;format="Camel"$.scala
│   ├── $if(shared.truthy)$shared$endif$
│   │   └── src
│   │       └── main
│   │           └── scala
│   │               └── $organization__packaged$
│   │                   └── package.scala
│   └── project
│       ├── build.properties
│       └── plugins.sbt
├── build.sbt
├── default.properties
└── project
    ├── build.properties
    └── plugins.sbt

24 directories, 10 files

```

**Comment by vptheron** [comment]
@notmu Thanks for the example, but it doesn't seem to work for me.  I see some strange behavior.  I have this layout:

```
src/main/g8
├── resources
 |   ├── $if(use_sql.truthy)$sql_scripts$endif$
 |    |          |-- db_script.sql
...
```

If I set `use_sql` to `yes` during project creation I see the expected layout:  `resources/sql_scripts/db_script.sql`.  However, if I use `no` to set the value of `use_sql` I see the following layout in the output project: `resources/db_script.sql`.  The file `db_script.sql` still gets included in the project, directly under `resources`.

Am I misunderstanding your example, or doing something wrong?

**Comment by taraspogrebniak** [comment]
@vptheron I have the same behavior as yours. the example from @notmu doesn't work for me

**Comment by VillenaDeveloper** [comment]
Same behaviour and It seem still no solution to this issue. It is impossible to delete files even with conditional creation for the root folder of the file. :(

**Comment by naxhh** [comment]
It does work if you do the same for the sub-files... it can be tediou...

**Comment by eugeniyk** [comment]
@eed3si9n why this issue was closed? I still see incorrect behavior for false cases

**Comment by ihostage** [comment]
@eugeniyk Because this issue was fixed in #432 
Can you share a repository that reproduces a problem?

**Comment by eugeniyk** [comment]
@ihostage I see, sorry for disturb
actually my bad - I was using sbt which was older version since it was installed long time ago (1.3.3)
running as `sbt -sbt-version 1.5.3 new ...` solves the problem
